enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -1589,6 +1596,77 @@ class AdjointGenerator
     ((DiffeGradientUtils *)gutils)->setDiffe(val, dif, Builder);
   }
 
+  /// Unwraps a vector derivative from its internal representation and applies a
+  /// function f to each element. Return values of f are collected and wrapped.
+  template <typename Func>
+  Value *applyChainRule(Type *diffType, Value *diff1, Value *diff2,
```

Comment:
Could this be renamed something like `vectorizeFwdRule` since this looks like it can only be used in forward mode?

Can the number of Values themselves be templated (rather than separate functions)?

Also, some of these functions aren't needed for bitcast, can you wait to propose them until a PR which uses them

Reply:
The function can also be used in reverse mode. It is used in erf for example.

Reply:
I'm not seeing this. Do you still need to push or am I missing something?

Reply:
Its in the vector-mode branch.